pkg/sqlite/query.go

```diff
@@ -1342,9 +1342,10 @@ func (s *SQLQuerier) listBundleChannels(ctx context.Context, bundleName string)
 // PackageFromDefaultChannelHeadBundle returns the package name if the provided bundle is the head of its default channel.
 func (s *SQLQuerier) PackageFromDefaultChannelHeadBundle(ctx context.Context, bundle string) (string, error) {
 	packageFromDefaultChannelHeadBundle := `
-	SELECT package_name FROM package 
-	INNER JOIN channel ON channel.name = package.default_channel
-	WHERE channel.head_operatorbundle_name = (SELECT name FROM operatorbundle WHERE bundlepath=? LIMIT 1) `
+	SELECT package_name FROM package, channel 
+	WHERE channel.package_name == package.name
+	AND package.default_channel == channel.name
+	AND channel.head_operatorbundle_name = (SELECT name FROM operatorbundle WHERE bundlepath=? LIMIT 1) `
```

Comment:
See that it was returning the package when the bundle is HEAD of channel but the channel is NOT the default channel of the package, e.g:

```
SELECT package_name FROM package 
	INNER JOIN channel ON channel.name = package.default_channel
	WHERE channel.head_operatorbundle_name = (SELECT name FROM operatorbundle WHERE bundlepath='quay.io/openshift-community-operators/apicurio-registry@sha256:239525b1a4a7030fb0184bea77a1e56b9bb652bbbb78e3facdedf829ee2f7e9d' LIMIT 1)
```

<img width="750" alt="Screen Shot 2021-09-16 at 16 32 57" src="https://user-images.githubusercontent.com/7708031/133641451-c53dbdd4-a275-489e-9624-991447fcfa5b.png">

The select with inner join was wrong.

Reply:
Same thing here, the inner join was missing a compare on `channel.package_name = package.name`, this looks good.